zooniverse/front-end-monorepo
# Classifier: replace SVGPoint with DOMPoint

## PR Overview

Package: `lib-classifier`
Affects: drawing tools
Closes #2268 

This PR replaces [SVGPoint](https://developer.mozilla.org/en-US/docs/Web/API/SVGPoint) with the funky fresh face of [DOMPoint](https://developer.mozilla.org/en-US/docs/Web/API/DOMPoint), which is so funky fresh that it's barely documented.

- But trust me, DOMPoint does indeed have `matrixTransform()`, and it _shooould_ work on all major modern browsers.
- Please see #2268 for full details.

### Testing

Test Project: basically any single-image workflow with a Drawing Task.
Test URL: https://localhost:8080/?env=production&project=1292&workflow=18372 , or I Fancy Cats work too

Testing steps:
- Switch to Move mode. Drag to pan the view on the Subject Image.
- Switch to Annotation mode. Draw a Point, Ellipse, or Rectangle. 
- Drag these marks to move them.
- For Ellipse or Rectangle, drag their "resize" handles to resize. 
- Rotate the image to test that drags continue to work with rotation.
- Drag-to-move and drag-to-resize the Points/Ellipses/Rectangles again.

Expectations for these actions should be pretty clear cut.

⚠️ NOTE: drag-to-pan (the view of the Subject Image) does NOT work when the image is rotated. This is documented in #2273

⚠️ NOTE: I noticed that with Rotating Rectangle, the resize handles don't work too well when the rectangle is rotated. This is unique to the Rotating Rectangle and I'm going to take a look at that separately.

### Status

Ready for review!
